class-components/src/App.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
+import './App.css';
+import TopSection from './components/TopSection/TopSection';
+import BottomSection from './components/BottomSection/BottomSection';
+import useLocalStorage from './hooks/useLocalStorage/useLocalStorage';
+import NotFound from './components/NotFound/NotFound';
+
+const App: React.FC = () => {
+  const [searchTerm, setSearchTerm] = useLocalStorage('searchTerm', '');
+
+  const handleSearch = (searchTerm: string) => {
+    setSearchTerm(searchTerm);
+  };
+
+  const handleError = () => {
+    throw new Error('Test Error! Test error!');
+  };
+
+  return (
+    <Router>
+      <div className='app'>
+        <Routes>
+          <Route
+            path='/'
+            element={
+              <React.Fragment>
+                <TopSection onSearch={handleSearch} />
+                <button onClick={handleError} className='btn'>
+                  Throw Error
+                </button>
+                <BottomSection searchTerm={searchTerm} />
```

Comment:
`TopSection` and `BottomSection` aren't too descriptive, it would be better to find more descriptive names for the components